src/sentry/web/frontend/auth_login.py

```diff
@@ -82,23 +237,286 @@ def get_auth_provider(self, organization_slug):
+            user_id=user.id,
+        )
+
+    def accept_invite_and_redirect_to_org(
+        self, request: Request, invite_helper: ApiInviteHelper
+    ) -> HttpResponseRedirect:
+        """
+        Accepts an invite on behalf of a user and redirects them to their org login
+        """
+        invite_helper.accept_invite()
+        org_slug = invite_helper.invite_context.organization.slug
+        self.determine_active_organization(request=request, organization_slug=org_slug)
+        response = self.redirect_to_org(request=request)
+        remove_invite_details_from_session(request=request)
+        return response
+
+    def get_redirect_url_for_successful_registration(self, request: Request) -> str:
+        """
+        Gets the redirect URL for a successfully submitted register form.
+        """
+        base_url = auth.get_login_redirect(request=request)
+        params = {"frontend_events": json.dumps({"event_name": "Sign Up"})}
+        return add_params_to_url(url=base_url, params=params)
+
+    def handle_login_form_submit(
+        self, request: Request, organization: Organization, **kwargs
+    ) -> HttpResponse:
+        """
+        Validates a completed login  form, redirecting to the next
+        step or returning the form with its errors displayed.
+        """
+        context = self.get_default_context(request=request, **kwargs)
+
+        login_form = AuthenticationForm(request, request.POST)
+
+        if self.is_ratelimited_login_attempt(request=request, login_form=login_form):
+            return self.get_ratelimited_login_form(request=request, login_form=login_form, **kwargs)
+        elif not login_form.is_valid():
+            context.update({"login_form": login_form, "op": "login"})
+            return self.respond_login(request=request, context=context, **kwargs)
+
+        user = login_form.get_user()
+        self._handle_login(request=request, user=user, organization=organization)
+        metrics.incr("login.attempt", instance="success", skip_internal=True, sample_rate=1.0)
+        return self.redirect_to_next_login_step(
+            request=request, user=user, organization=organization
+        )
+
+    def is_ratelimited_login_attempt(
+        self, request: Request, login_form: AuthenticationForm
+    ) -> bool:
+        """
+        Returns true if a user is attempting to login but is currently ratelimited.
+        """
+        from sentry import ratelimits as ratelimiter
+        from sentry.utils.hashlib import md5_text
+
+        attempted_login = request.POST.get("username") and request.POST.get("password")
+
+        return attempted_login and ratelimiter.is_limited(
+            "auth:login:username:{}".format(
+                md5_text(login_form.clean_username(value=request.POST["username"])).hexdigest()
+            ),
+            limit=5,
+            window=60,  # 5 per minute should be enough for anyone
+        )
+
+    def get_ratelimited_login_form(
```

Comment:
huh.
Do we need a completely separate helper for this?
Can we flatten this above as an optional arg in the generic return_login_form?

Reply:
it'd just be copy pasting the 12-line helper into the above function, i prefer it to be separated in here so its clear whats happening, do you feel strongly about having it above?